### ISSUE TITLE
Go get github.com/coredns/proxy fails

### ISSUE THREAD

**Issue by narengan** [coredns/proxy#2]
Error:
```
go get github.com/coredns/proxy
go: finding github.com/coredns/proxy latest
go: finding github.com/flynn/go-shlex latest
# github.com/coredns/proxy
../../../../pkg/mod/github.com/coredns/proxy@v0.0.0-20190312112157-36a505170348/proxy.go:107:22: state.ErrorMessage undefined (type request.Request has no field or method ErrorMessage)
```
This might be due to the following:
https://github.com/coredns/coredns/commit/f08f7e24d6a1ad9d41277e3dcf2d57edb7b9e00f#diff-960473a6998c1867436ca51f953284fe

**Comment by wwek** [comment]
same to you

**Comment by rampageX** [comment]
Ugly hack but...it's work...Just add:  `proxy:github.com/rampageX/proxy` to `plugin.cfg`.

> ./coredns -plugins
> Server types:
> dns
> 
> Caddyfile loaders:
> flag
> default
> 
> Other plugins:
> dns.alternate
> dns.any
> dns.auto
> dns.autopath
> dns.bind
> dns.cache
> dns.cancel
> dns.chaos
> dns.debug
> dns.dnssec
> dns.dnstap
> dns.erratic
> dns.errors
> dns.etcd
> dns.federation
> dns.file
> dns.forward
> dns.grpc
> dns.health
> dns.hosts
> dns.k8s_external
> dns.kubernetes
> dns.loadbalance
> dns.log
> dns.loop
> dns.metadata
> dns.nsid
> dns.pprof
> dns.prometheus
> **dns.proxy**
> dns.ready
> dns.reload
> dns.rewrite
> dns.root
> dns.route53
> dns.secondary
> dns.template
> dns.tls
> dns.trace
> dns.whoami
> on

**Comment by caiges** [comment]
@rampageX I gave that a try but got it failed with this output: https://gist.github.com/caiges/ed6d9215d41b3e7180f1cb555c645417

**Comment by miekg** [comment]
Will fix, but note this repo is best effort at best

**Comment by SuperQ** [comment]
@miekg Thanks, I understand. I'm happy to try fixing it. Just wondering if you want to restore the `ErrorMessage()` function or copy-n-paste fix it.

IMO, the reason for the original removal seems like the wrong way to do it. The function may be small/shallow, but it avoids copy-n-paste.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/proxy] Go get github.c..." ]
>@miekg Thanks, I understand. I'm happy to try fixing it. Just wondering if you want to restore the `ErrorMessage()` function or copy-n-paste fix it.
>
>IMO, the reason for the original removal seems like the wrong way to do it. The function may be small/shallow, but it avoids copy-n-paste.

I would like to keep request small, so ErrorMessage will not come back (it also 
wasn't used consistently)

**Comment by miekg** [comment]
fixed via #3
